homeassistant/components/zha/config_flow.py

```diff
@@ -164,9 +164,11 @@ async def async_step_zeroconf(
         """Handle zeroconf discovery."""
         # Hostname is format: livingroom.local.
         local_name = discovery_info.hostname[:-1]
+        radio_type = discovery_info.properties["radio_type"]
         node_name = local_name[: -len(".local")]
         host = discovery_info.host
-        device_path = f"socket://{host}:6638"
```

Comment:
this breaks the config flow on my tubeszb devices as they are now. Working on some new firmware that would work going forward (with another PR from me). but not sure how to make it work for both ways

Reply:
Which port is being returned with the current firmware?

Reply:
We should be able to get this from discovery info I think. I don’t remember why we hard coded this.

Reply:
it's not in there as currently piggy back off the esphome discovery info.

Reply:
I'll go hook one up and test to see if I can come up with a suggestion

Reply:
Thanks for the help!

Reply:
Should work now, but still needs some tests

Reply:
thanks, we should be able to get rid of the hard coding in the future.

Reply:
@tube0013 Can you test this to make sure it didn't break any of the `efr32` ones.  I don't have one of these to validate with

Reply:
@tube0013 Can you test this to make sure it didn't break any of the `efr32` ones.  I don't have one of these to validate with